[CR] Falling into deep water deals less damage

int-ua opened the pull request:
#### Summary
```SUMMARY: Balance "Falling into deep water deals much less damage"```

#### Purpose of change
Falling into deep water should not be as damaging as falling into shallow water.

#### Describe the solution
Taking into account if landing tile is deep water and character swimming skill.

#### Testing
Falling from 1 through 5, 10 and 20 levels with 0, 10 and 20 swimming skill.

Night-Pryanik commented:
Falling from a big height in the water almost equals falling on the concrete floor, especially if one is falling flat.

Siersha commented:
A compromise would be a linear scaling of damage taken based on your swimming skill. Consider it diving where 0 skill is belly flopping (full damage), and 10 skill is Olympic athlete (0 damage)

int-ua commented:
> Falling from a big height in the water almost equals falling on the concrete floor, especially if one is falling flat.

Well, yes. I just didn't think falling more than 2 z-levels into deep water is achievable in the current state of the game. But I agree that it has to be taken into account.

> A compromise would be a linear scaling of damage taken based on your swimming skill. Consider it diving where 0 skill is belly flopping (full damage), and 10 skill is Olympic athlete (0 damage)

https://en.wikipedia.org/wiki/High_diving
Looks like 50m is a reasonable limit for sustaining no damage with highest skill and on condition of being in control of your body. Which is about 10 z-levels in-game, no? Can't quickly find a canonical z-level height, please someone add a link.

Other links:
https://physics.stackexchange.com/questions/9059/jumping-into-water

https://www.physicsforums.com/threads/maximum-free-fall-dive-into-water.119781/

https://physics.stackexchange.com/questions/106808/why-is-jumping-into-water-from-high-altitude-fatal

CodeBandit commented:
> Looks like 50m is a reasonable limit for sustaining no damage with highest skill and on condition of being in control of your body. Which is about 10 z-levels in-game, no? Can't quickly find a canonical z-level height, please someone add a link.

Assuming climbing t the top of a 1 story house is on average 1 story high, then we can assume it is about 4m.
![image](https://user-images.githubusercontent.com/63547126/80308101-01e9c680-879b-11ea-9dea-39e80a712cd0.png)

int-ua commented:
> There are of course inconsistencies with this argument as radio towers can be over 400m tall, yet we don't need to go up 40 flights of stairs -hence why instead rely on swimming skill to mitigate damage and leave z level height as is

With 4.3 as average level height it's theoretically possible to fall from 86 meters height, where body reaches speed of almost 150 km/h, well above safe level even for most experienced divers. According to wiki, even at 75m height of Golden Gate bridge initial impact _survival_ rate is only at about 5%: https://en.wikipedia.org/wiki/Suicides_at_the_Golden_Gate_Bridge
So while I agree that swimming skill should be factored in it's not the only factor to account for.

I'm having trouble finding statistics for _minimum_ height at what injuries were recorded. Found code that might help: ICD-10-CM W16.112A: `Fall into natural body of water striking water surface causing other injury` but I still have to find any data sets containing useful data. Does anyone know any?

int-ua commented:
Fixed error that existed in the first version, replaced simple scaling with formula that accounts for swimming skill level. Not ideal, but results look somewhat reasonable. Any comments?

Night-Pryanik commented on src/player.cpp:
```diff
@@ -1273,8 +1273,12 @@ int player::impact( const int force, const tripoint &p )
         } else if( g->m.has_flag( TFLAG_SWIMMABLE, p ) ) {
-            // TODO: Some formula of swimming
-            effective_force /= 4;
+            int swim_skill = get_skill_level( skill_swimming );
```
```suggestion
            const int swim_skill = get_skill_level( skill_swimming );
```
